src/EditorFeatures/Core/Implementation/CommentSelection/AbstractCommentSelectionBase.cs

```diff
@@ -130,7 +130,13 @@ internal bool ExecuteCommand(ITextView textView, ITextBuffer subjectBuffer, TCom
             // Apply the text changes.
             using (var transaction = new CaretPreservingEditTransaction(title, textView, _undoHistoryRegistry, _editorOperationsFactoryService))
             {
-                document.Project.Solution.Workspace.ApplyTextChanges(document.Id, edits.TextChanges.Distinct(), CancellationToken.None);
+                using var edit = subjectBuffer.CreateEdit();
+                foreach (var change in edits.TextChanges.Distinct())
+                {
+                    edit.Replace(change.Span.ToSpan(), change.NewText);
+                }
+
+                edit.ApplyAndLogExceptions();
                 transaction.Complete();
```

Comment:
this is nto an appropriate change to make.  the benefit of going through the workspace is that we can ensure things like document-checkout work.  Please don't do this.